### PR TITLE
sqlint: init at 0.1.4

### PR DESCRIPTION
**Pull request by michalrus**
###### Motivation for this change

sqlint is useful (esp. with flycheck in Emacs)

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

But I need help. Currently, building `pg_query-0.9.2` fails, because it’s a native extension and needs a native library… How to provide one in `gemset.nix`?

### CODE REVIEW

**Comment by Mic92** [comment]
you need to add an override to `pkgs/development/ruby-modules/gem-config/default.nix` for `pg_query`

File: pkgs/development/ruby-modules/gem-config/default.nix [hunk 1]
```diff
@@ -129,6 +129,12 @@ in
     ];
   };
 
+  pg_query = attrs: {
+    postPatch = ''
+      ln -s ${libpg_query}/lib ${libpg_query.name}
```
Comment (michalrus) [inline pkgs/development/ruby-modules/gem-config/default.nix]: This doesn’t work… And is pretty awful, but take a look at https://github.com/lfittl/pg_query/blob/master/ext/pg_query/extconf.rb#L8-L11

What do, @Mic92? :-(

Comment (Mic92) [inline pkgs/development/ruby-modules/gem-config/default.nix]: the gem version in your project does not match the version on github: https://github.com/lfittl/pg_query/blob/v0.9.2/ext/pg_query/extconf.rb

Comment (michalrus) [inline pkgs/development/ruby-modules/gem-config/default.nix]: Oh, OK! I’ll look at this over the weekend.

What would we do, though, if we had two projects with each depending on a _different_ version of `pg_query`? We could depend only on _one_ version of `libpg_query` in this override, right? What then?

Comment (Mic92) [inline pkgs/development/ruby-modules/gem-config/default.nix]: How is this usually handled? @cstrahan

**Comment by ocharles** [comment]
@michalrus are you still working on this? I see it stalled earlier, could you help push it forward if you still need this work to be merged?

**Comment by michalrus** [comment]
@ocharles I’d *love* to have this util in my `PATH`, but I think I’m not the best person to grasp how to build (seemingly) non-trivial Ruby packages. :crying_cat_face: 

If anyone know how, please, help!

**Comment by ocharles** [comment]
Ok, I think it might be best to move this discussion to an issue, and perhaps trying to rally the troops via the mailing list.

**Comment by xeji** [comment]
superseded by #41901

**Pull request closed without merging**